In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import os

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [3]:
df=pd.read_csv("D:\ipl\IPL.csv")

C:\Users\karan\AppData\Local\Temp\ipykernel_8340\4278241017.py:1: DtypeWarning: Columns (28,29,30,31,43,46,47,48,51) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv("D:\ipl\IPL.csv")


In [4]:
df.shape

(278205, 64)

In [5]:
df.tail(5)

,Unnamed: 0,match_id,date,match_type,event_name,innings,batting_team,bowling_team,over,ball,ball_no,batter,bat_pos,runs_batter,balls_faced,bowler,valid_ball,runs_extras,runs_total,runs_bowler,runs_not_boundary,extra_type,non_striker,non_striker_pos,wicket_kind,player_out,fielders,runs_target,review_batter,team_reviewed,review_decision,umpire,umpires_call,player_of_match,match_won_by,win_outcome,toss_winner,toss_decision,venue,city,day,month,year,season,gender,team_type,superover_winner,result_type,method,balls_per_over,overs,event_match_no,stage,match_number,team_runs,team_balls,team_wicket,new_batter,batter_runs,batter_balls,bowler_wicket,batting_partners,next_batter,striker_out
278200,277951,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,2,19.2,Shashank Singh,6,0,1,JR Hazlewood,1,0,0,0,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,162,116,7,NaN,39,26,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278201,277952,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,3,19.3,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,168,117,7,NaN,45,27,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278202,277953,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,4,19.4,Shashank Singh,6,4,1,JR Hazlewood,1,0,4,4,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,172,118,7,NaN,49,28,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278203,277954,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,5,19.5,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,178,119,7,NaN,55,29,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278204,277955,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,6,19.6,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,184,120,7,NaN,61,30,0,"('KA Jamieson', 'Shashank Singh')",NaN,False


In [6]:
df['match_id'].nunique()

1169

1. the data is of 1169 ipl matches from 2008-2025
2. average matches per season is 69

In [20]:
df.isnull().sum()

Unnamed: 0                0
match_id                  0
date                      0
match_type                0
event_name                0
innings                   0
batting_team              0
bowling_team              0
over                      0
ball                      0
ball_no                   0
batter                    0
bat_pos                   0
runs_batter               0
balls_faced               0
bowler                    0
valid_ball                0
runs_extras               0
runs_total                0
runs_bowler               0
runs_not_boundary         0
extra_type           263072
non_striker               0
non_striker_pos           0
wicket_kind          264382
player_out           264382
fielders             268192
runs_target          144302
review_batter        277333
team_reviewed        277333
review_decision      277333
umpire               277333
umpires_call              0
player_of_match           0
match_won_by              0
win_outcome         

there are plenty of missing value but most of them are maybe due to data error .we can resolve that using the map function

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278205 entries, 0 to 278204
Data columns (total 64 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Unnamed: 0         278205 non-null  int64  
 1   match_id           278205 non-null  int64  
 2   date               278205 non-null  object 
 3   match_type         278205 non-null  object 
 4   event_name         278205 non-null  object 
 5   innings            278205 non-null  int64  
 6   batting_team       278205 non-null  object 
 7   bowling_team       278205 non-null  object 
 8   over               278205 non-null  int64  
 9   ball               278205 non-null  int64  
 10  ball_no            278205 non-null  float64
 11  batter             278205 non-null  object 
 12  bat_pos            278205 non-null  int64  
 13  runs_batter        278205 non-null  int64  
 14  balls_faced        278205 non-null  int64  
 15  bowler             278205 non-null  object 
 16  va

In [22]:
df.duplicated().sum()

np.int64(0)

# data cleaning

In [7]:
train_df=df.copy()

In [8]:
train_df.tail()

,Unnamed: 0,match_id,date,match_type,event_name,innings,batting_team,bowling_team,over,ball,ball_no,batter,bat_pos,runs_batter,balls_faced,bowler,valid_ball,runs_extras,runs_total,runs_bowler,runs_not_boundary,extra_type,non_striker,non_striker_pos,wicket_kind,player_out,fielders,runs_target,review_batter,team_reviewed,review_decision,umpire,umpires_call,player_of_match,match_won_by,win_outcome,toss_winner,toss_decision,venue,city,day,month,year,season,gender,team_type,superover_winner,result_type,method,balls_per_over,overs,event_match_no,stage,match_number,team_runs,team_balls,team_wicket,new_batter,batter_runs,batter_balls,bowler_wicket,batting_partners,next_batter,striker_out
278200,277951,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,2,19.2,Shashank Singh,6,0,1,JR Hazlewood,1,0,0,0,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,162,116,7,NaN,39,26,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278201,277952,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,3,19.3,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,168,117,7,NaN,45,27,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278202,277953,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,4,19.4,Shashank Singh,6,4,1,JR Hazlewood,1,0,4,4,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,172,118,7,NaN,49,28,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278203,277954,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,5,19.5,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,178,119,7,NaN,55,29,0,"('KA Jamieson', 'Shashank Singh')",NaN,False
278204,277955,1473511,2025-06-03,T20,Indian Premier League,2,Punjab Kings,Royal Challengers Bengaluru,19,6,19.6,Shashank Singh,6,6,1,JR Hazlewood,1,0,6,6,False,NaN,KA Jamieson,9,NaN,NaN,NaN,191.0,NaN,NaN,NaN,NaN,False,KH Pandya,Royal Challengers Bengaluru,6 runs,Punjab Kings,field,"Narendra Modi Stadium, Ahmedabad",Ahmedabad,3,6,2025,2025,male,club,NaN,NaN,NaN,6,20,Unknown,Final,Unknown,184,120,7,NaN,61,30,0,"('KA Jamieson', 'Shashank Singh')",NaN,False


droping the unnamed :0 ,match type, event name : as these are same for all the mathes.and  ipl are of 20 overs

In [9]:
train_df.drop(columns=['Unnamed: 0','match_type','event_name'],inplace=True)

In [10]:
train_df.head()

,match_id,date,innings,batting_team,bowling_team,over,ball,ball_no,batter,bat_pos,runs_batter,balls_faced,bowler,valid_ball,runs_extras,runs_total,runs_bowler,runs_not_boundary,extra_type,non_striker,non_striker_pos,wicket_kind,player_out,fielders,runs_target,review_batter,team_reviewed,review_decision,umpire,umpires_call,player_of_match,match_won_by,win_outcome,toss_winner,toss_decision,venue,city,day,month,year,season,gender,team_type,superover_winner,result_type,method,balls_per_over,overs,event_match_no,stage,match_number,team_runs,team_balls,team_wicket,new_batter,batter_runs,batter_balls,bowler_wicket,batting_partners,next_batter,striker_out
0,335982,2008-04-18,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,1,0.1,SC Ganguly,1,0,1,P Kumar,1,1,1,0,False,legbyes,BB McCullum,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,1,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
1,335982,2008-04-18,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,2,0.2,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
2,335982,2008-04-18,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,3,0.3,BB McCullum,2,0,0,P Kumar,0,1,1,1,False,wides,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
3,335982,2008-04-18,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,3,0.3,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,3,0,NaN,0,2,0,"('BB McCullum', 'SC Ganguly')",NaN,False
4,335982,2008-04-18,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,4,0.4,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,4,0,NaN,0,3,0,"('BB McCullum', 'SC Ganguly')",NaN,False


we are also deleting the date column as it is already extracted into day /month /year

In [11]:
train_df.drop('date',axis=1,inplace=True)

In [12]:
train_df.head()

,match_id,innings,batting_team,bowling_team,over,ball,ball_no,batter,bat_pos,runs_batter,balls_faced,bowler,valid_ball,runs_extras,runs_total,runs_bowler,runs_not_boundary,extra_type,non_striker,non_striker_pos,wicket_kind,player_out,fielders,runs_target,review_batter,team_reviewed,review_decision,umpire,umpires_call,player_of_match,match_won_by,win_outcome,toss_winner,toss_decision,venue,city,day,month,year,season,gender,team_type,superover_winner,result_type,method,balls_per_over,overs,event_match_no,stage,match_number,team_runs,team_balls,team_wicket,new_batter,batter_runs,batter_balls,bowler_wicket,batting_partners,next_batter,striker_out
0,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,1,0.1,SC Ganguly,1,0,1,P Kumar,1,1,1,0,False,legbyes,BB McCullum,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,1,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
1,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,2,0.2,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
2,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,3,0.3,BB McCullum,2,0,0,P Kumar,0,1,1,1,False,wides,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
3,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,3,0.3,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,3,0,NaN,0,2,0,"('BB McCullum', 'SC Ganguly')",NaN,False
4,335982,1,Kolkata Knight Riders,Royal Challengers Bangalore,0,4,0.4,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,4,0,NaN,0,3,0,"('BB McCullum', 'SC Ganguly')",NaN,False


inning is defining the which club is playing first/bating  and which club is bowling  

replacing the names of batting team into the sort form (example "Kolkata Knight Riders"-kkr) and so on 

In [13]:
train_df['batting_team'].unique()

array(['Kolkata Knight Riders', 'Royal Challengers Bangalore',
       'Chennai Super Kings', 'Kings XI Punjab', 'Rajasthan Royals',
       'Delhi Daredevils', 'Mumbai Indians', 'Deccan Chargers',
       'Kochi Tuskers Kerala', 'Pune Warriors', 'Sunrisers Hyderabad',
       'Rising Pune Supergiants', 'Gujarat Lions',
       'Rising Pune Supergiant', 'Delhi Capitals', 'Punjab Kings',
       'Lucknow Super Giants', 'Gujarat Titans',
       'Royal Challengers Bengaluru'], dtype=object)

some of the teams are have 2 value so we will convert these into lower case teams

In [14]:
train_df['batting_team'].nunique()

19

In [15]:
batting_dic={
            'Kolkata Knight Riders':'KKR',
            'Royal Challengers Bangalore':'RCB',
            'Chennai Super Kings':'CSK',
            'Kings XI Punjab':'KXP',
            'Rajasthan Royals':'RR',
            'Delhi Daredevils':'DC',
            'Mumbai Indians':'MI',
            'Deccan Chargers':'DEC',
            'Kochi Tuskers Kerala':'KTK',
            'Pune Warriors':"PW",
            'Sunrisers Hyderabad':'SRH',
            'Rising Pune Supergiants':'RPS',
            'Gujarat Lions':'GL',
            'Rising Pune Supergiant':'RPS',
            'Delhi Capitals':'DC',
            'Punjab Kings':'KXP',
            'Lucknow Super Giants':'LSG',
            'Gujarat Titans':'GT',
            'Royal Challengers Bengaluru':'RCB'}

In [17]:
# train_df.apply(lambda x : x .replace(batting_dic))
# this process can be slow so insted of using lamda we are going to use the replace function of pandas which is faster or the map function 
# Create a list of columns that contain team names
team_cols = ['batting_team', 'bowling_team'] 

# Option A: Using replace (Great for whole DataFrames or specific columns)
# Strip spaces and replace names for both columns efficiently
for col in team_cols:
    train_df[col] = train_df[col].str.strip().replace(batting_dic)# we added the strip fuction so that the there is no extrea space in
# the daataset which can cause problem in the future

In [24]:
train_df.head(8)

,match_id,innings,batting_team,bowling_team,over,ball,ball_no,batter,bat_pos,runs_batter,balls_faced,bowler,valid_ball,runs_extras,runs_total,runs_bowler,runs_not_boundary,extra_type,non_striker,non_striker_pos,wicket_kind,player_out,fielders,runs_target,review_batter,team_reviewed,review_decision,umpire,umpires_call,player_of_match,match_won_by,win_outcome,toss_winner,toss_decision,venue,city,day,month,year,season,gender,team_type,superover_winner,result_type,method,balls_per_over,overs,event_match_no,stage,match_number,team_runs,team_balls,team_wicket,new_batter,batter_runs,batter_balls,bowler_wicket,batting_partners,next_batter,striker_out
0,335982,1,KKR,RCB,0,1,0.1,SC Ganguly,1,0,1,P Kumar,1,1,1,0,False,legbyes,BB McCullum,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,1,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
1,335982,1,KKR,RCB,0,2,0.2,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,1,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
2,335982,1,KKR,RCB,0,3,0.3,BB McCullum,2,0,0,P Kumar,0,1,1,1,False,wides,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,2,0,NaN,0,1,0,"('BB McCullum', 'SC Ganguly')",NaN,False
3,335982,1,KKR,RCB,0,3,0.3,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,3,0,NaN,0,2,0,"('BB McCullum', 'SC Ganguly')",NaN,False
4,335982,1,KKR,RCB,0,4,0.4,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,4,0,NaN,0,3,0,"('BB McCullum', 'SC Ganguly')",NaN,False
5,335982,1,KKR,RCB,0,5,0.5,BB McCullum,2,0,1,P Kumar,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,2,5,0,NaN,0,4,0,"('BB McCullum', 'SC Ganguly')",NaN,False
6,335982,1,KKR,RCB,0,6,0.6,BB McCullum,2,0,1,P Kumar,1,1,1,0,False,legbyes,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,3,6,0,NaN,0,5,0,"('BB McCullum', 'SC Ganguly')",NaN,False
7,335982,1,KKR,RCB,1,1,1.1,BB McCullum,2,0,1,Z Khan,1,0,0,0,False,NaN,SC Ganguly,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,False,BB McCullum,Kolkata Knight Riders,140 runs,Royal Challengers Bangalore,field,M Chinnaswamy Stadium,Bangalore,18,4,2008,2007/08,male,club,NaN,NaN,NaN,6,20,1,Unknown,Unknown,3,7,0,NaN,0,6,0,"('BB McCullum', 'SC Ganguly')",NaN,False
